ext_mongodb.php

```diff
@@ -715,17 +720,43 @@ function __debugInfo() : array;
 {
 	use DenySerialization;
 
-	public function __construct(private string $code, private ?mixed $scope = NULL)
+	private $code;
+
+	public function __construct(string $code, private ?mixed $scope = NULL)
 	{
+		if ( strstr( $code, "\0" ) !== false )
+		{
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "Code cannot contain null bytes" );
+		}
+		$this->code = $code;
 	}
 
 	public function __debugInfo() : array
 	{
 		return [
-			'javascript' => $this->code,
+			'code' => $this->code,
 			'scope' => (object) $this->scope
 		];
 	}
+
+	public function getCode() : string
+	{
+		return $this->code;
+	}
+
+	public function getScope() : mixed
+	{
+		if ( $this->scope !== NULL )
+		{
+			return (object) $this->scope;
```

Comment:
It is not round tripping anything. It takes it right from the value set through the constructor. The casting was necessary to pass one of your tests (the one you mentioned)—I really just copied it over.